command/genesis/utils.go

```diff
@@ -246,6 +246,35 @@ func ReadValidatorsByPrefix(dir, prefix string) ([]*validator.GenesisValidator,
 	return validators, nil
 }
 
+// ReadValidatorsByPrefix reads validators secrets on a given root directory and with given folder prefix
```

Comment:
I would remove this function completely, and just modify the existing `ReadValidatorsByPrefix`, and add this to it:

```go
stake, exists := stakeInfos[types.Address(account.Ecdsa.Address())]
if !exists {
    stake = big.NewInt(0)
}

validators[i],Stake = stake
```